### ISSUE TITLE
java.lang.UnsatisfiedLinkError: C:\Users\...\AppData\Local\Continuum\anaconda3\Lib\site-packages\jep\jep.dll: Can't find dependent libraries

### ISSUE THREAD

**Issue by SaintNazaire** [shadaj/scalapy#6]
## **System information**

- Windows 10 professional
- Scala 2.12.3 (built-in), 2.10.6 (built-in), 2.11.11 (built-in)
- sbt 1.1.2 (local 0.13.13 as per scalapy's build.sbt specs)
- Java 1.8 JRE 1.8.0_171
- Java 8 JDK 171
- Python 3.6.4 (Conda 4.4.10, Anaconda 1.6.9 64x bits)
- jep 3.7.1
- Eclispe Oxygen.3a (4.7.3a) build 20180405-1200
- PyDev 6.3.3.201805051638
- build.sbt config:
```
libraryDependencies += "me.shadaj" %% "scalapy" % "0.2.0"
fork in run := true
javaOptions += "-Djava.library.path=/Users/.../AppData/Local/Continuum/anaconda3/Lib/site-packages/jep"
```

## **Two approaches**

**Approach 1 - pointing to the jar file from properties in Eclipse**

Click right on project > Properties > Java Build Path > Libraries > Add External JARs

**Approach 2 - setting-up a lib folder with jep.dll file**

Copy and pasting Conda's built of jep.dll

By the way inspecting both referenced librairies indeed yield no trace of java.lang yet I read from the Scala documentation that java.lang is imported by default (scroll down to bottom of the following [page](https://docs.scala-lang.org/tour/packages-and-imports.html) linked)

## **Command triggering issue & relevant error log**

Run selection in Scala Interpreter and same from command line

> java.lang.UnsatisfiedLinkError: C:\Users\...\AppData\Local\Continuum\anaconda3\Lib\site-packages\jep\jep.dll: Can't find dependent libraries
>   at java.lang.ClassLoader$NativeLibrary.load(Native Method)
>   at java.lang.ClassLoader.loadLibrary0(Unknown Source)
>   at java.lang.ClassLoader.loadLibrary(Unknown Source)
>   at java.lang.Runtime.loadLibrary0(Unknown Source)
>   at java.lang.System.loadLibrary(Unknown Source)
>   at jep.Jep.<clinit>(Jep.java:217)
>   ... 48 elided
> java.lang.UnsatisfiedLinkError: C:\Users\GB5026\AppData\Local\Continuum\anaconda3\Lib\site-packages\jep\jep.dll: Can't find dependent libraries
>   at java.lang.ClassLoader$NativeLibrary.load(Native Method)
>   at java.lang.ClassLoader.loadLibrary0(Unknown Source)
>   at java.lang.ClassLoader.loadLibrary(Unknown Source)
>   at java.lang.Runtime.loadLibrary0(Unknown Source)
>   at java.lang.System.loadLibrary(Unknown Source)
>   at jep.Jep.<clinit>(Jep.java:217)
>   ... 48 elided

**Comment by SaintNazaire** [comment]
All else failed but I got one solution working which is to add jep as a managed dependency approach in build.sbt

```
libraryDependencies += "black.ninia" % "jep" % "3.7.0"
```

Also make sure that jep installed in python is 3.7.0

`pip install jep=3.7.0`

Then in Properties > Java Build Path > Librairies > jep-3.7.0.jar > Native librairy location, set the location path to the .dll, typically something like:

`C:/Users/.../Anaconda3/Lib/site-packages/jep`

Do not forget to add the following paths to the Windows environment variables %PATH%, note this links to jep but also the python librairies:

```
C:/Users/.../Anaconda3/Lib/site-packages/jep
C:/Users/.../Anaconda3/
C:/Users/.../Anaconda3/DLLs
```
Now do not forget to close restart Eclipse (or the Windows environment variables won't be reset), then 

```
sbt reload
sbt publishLocal
```

This should be a functional workaround to fix ninia/jep#141

**Comment by shadaj** [comment]
Closing since ScalaPy no longer uses Jep
